# Topics Advertised
TrajectoryTopic:            '/wtr/trajectory'
WayPointsTopic:             '/wtr/waypoints'
RouteTopic:                 '/wtr/route'
AccTopic:                   '/wtr/acc'

# Frame Name
FrameName:                  'world'

# Traj Plan Params
WeightT:                     1024.0

WeightAcc:                   16.0

WeightJerk:                  0.4

MaxAccRate:                  1.50

MaxVelRate:                  2

Iterations:                  23

Epsilon:                     0.02


#process1
# Number of Points 
PointNumber:                   3

#process2
#Different Number of Points in every part .cpp文件中maxPointParts设置20,最多区间小于20即可
PointArray:                   [3,3,4,4]
#Maximum segment  此值小于等于PointArray元素个数即可
MaxParts:                       4

#process3
#The point coordinates to be passed in
                             #x       #y      #z
PointSet:                 [ 0.0,      0.0,    0.0,           
                            1,      2,        0.0,            
                            2,      3,        0.0,
                            3,      4,        0.0,
                            4,      5,        0.0]           
MaxPointSetNumber:              15         #3*3,PointSet元素个数，即坐标点数*3

#process4(该流程ER取环射环更方便，缺少避障功能同时对RR还需更改)
#Set the coordinate points of multiple segments
                             #x       #y      #z
PointSet_:                 [0,      0,      0.0,           
                            1,      2,      0.0,            
                            2,      3,      0.0,

                            3,      4,      0.0,
                            4,      5,      0.0,
                            5,      6,      0.0,
                            
                            6,      7,      0.0,
                            7,      8,      0.0,
                            8,      9,      0.0,
                            9,      10,     0.0,

                            10,     11,     0.0,
                            11,     12,     0.0,
                            12,     13,     0.0,
                            13,     14,     0.0
                            ]   
MaxPointSetNumber_:              42       #上面一维数组元素总数
PointArray_:                 [3,3,4,4]    #每个区段的点数(能力有限，应与坐标元素数组间隔一一对应)
MaxParts_:                       4        #此值小于等于PointArray_元素个数即可

#process5(based on process4，主要用于RR,扫出地图之后结合添加避障功能)
#射环区                     
FireZoneA:                     [1.5,5.5,0]
FireZoneB:                     [5.0,-5.5,0]
FireZoneC:                     [1.5,2,0]
StartFireZone:                     [1.5,5.5,0]
StartMoveZone:                    [0.5,0.5,0]
expand_factor:                      2     #一倍膨胀对应5个像素
point_pace:                         20    

#取环区(待添加)

#自主决策，依赖视觉.先简单决策应该射哪一个环，能实现的话考虑通过哪种方式获胜(见文档,参数后续添加)
auto_decesion:
  Chasis:
    #底盘xyz对map原点偏移量
    ChasisDxyz:                       [0,0,0]
  LidarPart:
    #开机后雷达运行拖延时间
    LidarDecayTime:                 15
    #箱子中心拓展范围(环的最大直径25cm)
    DeltaX:                         0.26
    DeltaY:                         0.26
    DeltaZ:                         0.2
    #雷达xyz对原点(map零点)偏移量：
    Lidar2mapDxyz:                    [0.3,0.2,0.1]

  #每个发射区对应九个柱子命中率
  ZoneA:
    HitRate:                        [0,0,0,  #己方一型柱
                                    0,0,0,   #对方一型柱
                                    0,0,0,0,   #二型柱 
                                    0]         #三型柱
  ZoneB:   
    HitRate:                        [0,0,0,  
                                    0,0,0,   
                                    0,0,0,0,   
                                    0]         
  ZoneC:
    HitRate:                        [0,0,0,  #己方一型柱
                                    0,0,0,   #对方一型柱
                                    0,0,0,0,   #二型柱 
                                    0]         #三型柱                               
  #射中分数
  HitPoint:                        [10,10,10,
                                    25,25,25,
                                    30,30,30,30,
                                    70] 
  #每个柱子托盘位置(柱子序号在pdf中从上到下，从右到左，按类型分)
  PillarLocation:                  [2.75,-3.2,0.8,  2.75,0,0.8,  2.75,3.2,0.8,
                                    9.15,-3.2,0.8,  9.15,0,0.8,  9.15,3.2,0.8,  
                                    4.65,-1.3,1.0,  4.65,1.3,1.0,  7.25,-1.3,1.0,  7.25,1.3,1.0,  
                                    6,0,1.6]
  #柱子状态(0表示没有柱子,1是自己的,2是对面的)
  OverallSituation:             [0,0,0,
                                0,0,0,
                                0,0,0,0,
                                0]
  TargetPillar:                     -1
  #s时间
  Time:
    #转移时间:
    TransTime:            [1,2,3,4,5,6]
      # AtoB:                 1
      # AtoC:                 1
      # BtoA:                 1
      # BtoC:                 1
      # CtoA:                 1
      # ctoB:                 1
    #调整时间：(到达之后到正式把环发射出去所需的全部时间)
    AdjustTime:
      ZoneA:              [0,0,0,
                            0,0,0,
                            0,0,0,0,
                            0]
      ZoneB:              [0,0,0,
                            0,0,0,
                            0,0,0,0,
                            0]
      ZoneC:              [0,0,0,
                            0,0,0,
                            0,0,0,0,
                            0]

